mp4parse/src/lib.rs

```diff
@@ -565,6 +571,32 @@ fn read_moov<T: Read>(f: &mut BMFFBox<T>, context: &mut MediaContext) -> Result<
     Ok(())
 }
 
+fn read_mvex<T: Read>(src: &mut BMFFBox<T>) -> Result<MovieExtendsBox> {
+    let mut iter = src.box_iter();
+    let mut duration: Option<u64> = None;
+    while let Some(mut b) = try!(iter.next_box()) {
+        match b.head.name {
+            BoxType::MovieExtendsHeaderBox => {
+                duration = try!(read_mehd(&mut b));
+            },
+            _ => try!(skip_box_content(&mut b)),
+        }
+    }
+    Ok(MovieExtendsBox {
+        fragment_duration: duration.map_or(None, |x| { Some(MediaScaledTime(x)) }),
+    })
+}
+
+fn read_mehd<T: Read>(src: &mut BMFFBox<T>) -> Result<Option<u64>> {
```

Comment:
`mehd` always has a fragment duration, so you don't need to return an Option here. A simpler `Result<u64>` or `Result<MediaScaledTime>` would work. This complicates the way you update duration, but I think it's better to have a temporary there and keep the semantics of the function signature correct.

e.g.
```rust
let mut fragment_duration = None;
...
            let duration = try!(read_mehd(&mut b));
            fragment_duration = Some(duration);
...
Ok(MovieExtendsBox {
    fragment_duration: fragment_duration,
})
```